src/coins.cpp

```diff
@@ -1003,6 +1003,19 @@ CAmount CCoinsViewCache::GetValueIn(const CTransaction& tx) const
     return nResult;
 }
 
+CAmount CCoinsViewCache::GetTransparentValueIn(const CTransaction& tx) const
```

Comment:
`GetValueIn` can then be implemented as `GetTransparentValueIn(tx) + tx.GetShieldedValueIn()`

Reply:
Given that this is dependent on circuit correctness and soundness, we probably want to assert that `tx.GetShieldedValueIn()` returns 0 for coinbase (and probably in that function rather than here).

Reply:
I don't think this suggestion was followed.

Reply:
Fixed in the "Apply suggested changes" commit.